ethers-core/src/types/transaction/response.rs

```diff
@@ -344,9 +344,13 @@ impl Decodable for Transaction {
             true => Ok(Some(rlp.data()?.into())),
             false => Ok(None),
         }?;
-        let rest = rlp::Rlp::new(
-            rlp.as_raw().get(1..).ok_or(DecoderError::Custom("no transaction payload"))?,
-        );
+
+        let rest = match txn.transaction_type {
+            Some(_) => rlp::Rlp::new(
+                rlp.as_raw().get(1..).ok_or(DecoderError::Custom("no transaction payload"))?,
+            ),
+            None => rlp.to_owned(),
+        };
```

Comment:
since none of the arms are used here I'd recommend a simple `if.txn.transaction_type.is_some(){}else{};` here

Reply:
Updated!